### PR TITLE
Added condition to hide maxLength onBlur

### LINKED ISSUE #1821: Provide prop to hide maxLength when input field is not focused
Showing character count even if the field is not focused: https://vimeo.com/861552006/4b993162d8

### PR DESCRIPTION
**Pull request by neenu-chacko**
- Fixes #1821 
Demo: https://neenu-chacko.neetorecord.com/watch/ec8b094a-2b60-45fa-8db9-8014fcf12315

- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by praveen-murali-ind** [comment]
@neenu-chacko Let’s proceed with the bottom alignment for now.

Screenshot 2023-10-18 at 9 02 18 PM

File: src/components/formik/Input.jsx [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+          error={meta.touched ? meta.error : ""}
+          isFormikInputFocused={isFormikInputFocused}
+          {...rest}
+          onBlur={() => setIsFormikInputFocused(false)}
+          onFocus={() => setIsFormikInputFocused(true)}
+        />
+      )}
+    </Field>
+  );
+});
```
Comment (ajmaln) [inline src/components/formik/Input.jsx:26]: As discussed let's revert this change

File: src/components/Input.jsx [hunk 2]
```diff
@@ -108,6 +100,8 @@ const Input = forwardRef(
               [errorId]: !!error,
               [helpTextId]: helpText,
             })}
+            onBlur={() => setIsInputFocused(false)}
+            onFocus={() => setIsInputFocused(true)}
```
Comment (ajmaln) [inline src/components/Input.jsx:104]: Let's make sure that the onBlur and onFocus from the host app is also executed.

**Review by ajmaln — COMMENTED** [review]
@neenu-chacko _a Please check

**Comment by praveen-murali-ind** [comment]
@neenu-chacko Let’s revert the style changes in neetoUI, as they affect other inputs. 

For this particular use case, we can override the default styling on the project side to make it the bottom aligned as discussed.

Screenshot 2023-10-18 at 9 02 18 PM
